### ISSUE TITLE
[1.12.2] Can't get get new biome to spawn in nether

### ISSUE THREAD

**Issue by HiddenByNight** [superckl/BiomeTweaker#303]
I can't get this new biome to generate in the nether. The biome shows up in the output folder and appears in the list when I list biomes, but it won't generate.

```
Tweaker.setScriptStage("BIOME_REGISTRY")
Tweaker.createBiome("basalt_plains","minecraft:hell")
Tweaker.setWorld(-1)

basaltP = forBiomes("biometweaker:basalt_plains")

basaltP.removeAllSpawns("CREATURE")
basaltP.removeAllSpawns("MONSTER")
basaltP.addSpawn("cofh.thermalfoundation.entity.monster.EntityBasalz", "MONSTER", 100, 4, 4)

basaltP.set("temperature", 2.0)
basaltP.set("genWeight", 1000)
basaltP.set("skyColor", 4873389)
basaltP.set("fogColor", 3160944)
basaltP.set("name", "Basalt Plains")
basaltP.set("topBlock", "quark:basalt")
basaltP.set("fillerBlock", "quark:basalt")

basaltP.addDicTypes("NETHER")

Tweaker.setScriptStage("FINISHED_LOAD")
basaltP.addToGeneration("DESERT", 2000)
basaltP.addToGeneration("WARM", 2000)
basaltP.addToGeneration("ICY", 2000)
basaltP.addToGeneration("COOL", 2000)
```

[latest.log](https://github.com/superckl/BiomeTweaker/files/14668946/latest.log)
I also tried doing the same with every other mod uninstalled but that didn't fix it.

**Comment by Dinerell** [comment]
`basaltP.addSpawn("cofh.thermalfoundation.entity.monster.EntityBasalz", "MONSTER", 100, 4, 4)` that isn't how you add spawns. You have to use the resource location like `"cofh:basalz"` although I'm not sure which they actually have, normally you find the resource location by advanced tooltips through `F3+H` for blocks and items or in your case you can use `/ct entity_info` to look it up. Any error in the script prevents it from executing and there is no error log!

**Comment by HiddenByNight** [comment]
Thanks for the reply, but that doesn't seem to be the problem. I removed the three lines regarding creature spawning and the result is the same.

**Comment by Dinerell** [comment]
It could be that you need to change the Script Stage back to `FINISHED_LOAD` (default) before accessing the biome with `basaltP = forBiomes("biometweaker:basalt_plains")`. I'm not sure how it works for that but if I guess it biomes are registered at the end of the biome registration meaning you can't access the biome through the resource location at the time.

**Comment by HiddenByNight** [comment]
That's not it either
`Tweaker.setScriptStage("BIOME_REGISTRY")
Tweaker.createBiome("basalt_plains","minecraft:hell")
Tweaker.setWorld(-1)

Tweaker.setScriptStage("FINISHED_LOAD")

basaltP = forBiomes("biometweaker:basalt_plains")

basaltP.set("temperature", 2.0)
basaltP.set("genWeight", 1000)
basaltP.set("skyColor", 4873389)
basaltP.set("fogColor", 3160944)
basaltP.set("name", "Basalt Plains")
basaltP.set("topBlock", "quark:basalt")
basaltP.set("fillerBlock", "quark:basalt")

basaltP.addDicTypes("NETHER")

basaltP.addToGeneration("DESERT", 2000)
basaltP.addToGeneration("WARM", 2000)
basaltP.addToGeneration("ICY", 2000)
basaltP.addToGeneration("COOL", 2000)`

**Comment by Dinerell** [comment]
The last thing I can think of is that you need to `basaltP.addToGeneration("NETHER", 2000)`. In 1.12 there is only one biome for the Nether which resource location is `minecraft:hell`, so it could be that the overworld type don't even do anything. You could try removing `Tweaker.setWorld(-1)` or add an `#` in front of it to mark it as a comment and then try to find it in the overworld and if it exists, the generation types probably won't affect generation in the Nether.

**Comment by HiddenByNight** [comment]
It indeed spawns in the overworld. From what I understand, this mod can't be used to make changes in the nether, at least by itself. I found somewhat of a workaround tho:

I installed NetherEx and used it to get this biome to spawn. In saves/world file/config/netherex/nether_biomes/netherex there are four files which contain biome info. I made a copy of one of those and renamed it to "basalt_plains". Then I simply changed the first line to `"biome": "biometweaker:basalt_plains",. I don't know how much customization I can get out of this, but so far I was able to set blocks, entities and weight of the biome. So this looks like a promising method to make nether a proper dimension. Thanks again for your help.
